libraries/Microsoft.Bot.Configuration/BotConfigurationExtensions.cs

```diff
@@ -0,0 +1,153 @@
+        {
+            // If the parameter 'folder' is null or empty, throws an Exception
+            if (string.IsNullOrEmpty(folder))
+            {
+                throw new ArgumentNullException(nameof(folder));
+            }
+
+            var files = Directory.GetFiles(folder, "*.csproj", SearchOption.TopDirectoryOnly);
+
+            // Throws an error if there are two or more 'csproj' files
+            if (files.Length > 1)
+            {
+                throw new Exception("There could not be more than one '.csproj' file.");
+            }
+
+            try
+            {
+                // Gets the path to the first (and only) file with extension 'csproj'.
+                var file = files.FirstOrDefault();
+
+                // Checks if the file name contains any white space
+                return !file.Contains(" ");
+            }
+            catch (FileNotFoundException ex)
+            {
+                // If there's no file, throws an Exception
+                throw new FileNotFoundException("There isn't any \'.csproj\' in the specified folder", ex);
+            }
+        }
+
+        /// <summary>
+        /// Checks if the <see cref="LuisService"/> has its key defined.
+        /// </summary>
+        /// <param name="botConfiguration">BotConfiguration object to extend.</param>
+        /// <returns>Tuple with a boolean that indicates if a bot configuration file's Luis keys are present and its Error message.</returns>
+        public static Tuple<bool, string> ValidateLuisKeyExistence(this BotConfiguration botConfiguration)
+        {
+            // Get all the luis types in the service list
+            var luisServices = botConfiguration.Services.OfType(ServiceTypes.Luis);
+
+            // If there isn't any Luis service, returns false and the corresponding error message
+            if (luisServices == null)
+            {
+                return Tuple.Create(false, "There is no LUIS service in the bot configuration file.");
+            }
+
+            // Check if exists any LuisService without its key (SubscriptionKey)
+            var existEmptyLuis = luisServices.Any(x => string.IsNullOrWhiteSpace(((LuisService)x).SubscriptionKey));
+
+            // If there is at least one LuisService without its key, the process will return false
+            if (existEmptyLuis)
+            {
+                return Tuple.Create(false, "There is at least one LuisService without a key.");
+            }
+
+            // Get all the Dispatch (Luis) types in the service list
+            var dispatchServices = botConfiguration.Services.OfType(ServiceTypes.Dispatch);
+
+            // The dispatch service is not always included in the the bot configuration files.
+            // Therefore, it will only be validated if the bot configuration file contains at least one service of this type.
+            if (dispatchServices.Count() != 0)
```

Comment:
Again, `.Count()` is `O(N)`... you should really just do your `.Any` check from below right here.